### PR TITLE
Bump js to get fix from servo/rust-mozjs#237

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9606)

### CODE REVIEW

**Comment by nox** [comment]
```
  ▶ CRASH [expected OK] /_mozilla/mozilla/sequence-hole.html
  │
  │ thread 'ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'assertion failed: self.is_object()', /Users/nox/src/servo/.cargo/git/checkouts/rust-mozjs-ebb4917e843c0a11/master/src/jsval.rs:339
  │ stack backtrace:
  │    1:        0x103a609f8 - sys::backtrace::tracing::imp::write::h268482af15af0a80Y8t
  │    2:        0x103a62b55 - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.42271
  │    3:        0x103a62756 - panicking::default_handler::h97bc664f7e2bfeb5Ocy
  │    4:        0x103a4e716 - sys_common::unwind::begin_unwind_inner::h98c0975c9b6b2fa0Zbt
  │    5:        0x10337bdf7 - sys_common::unwind::begin_unwind::h223866613450113180
  │    6:        0x103386266 - jsval::JSVal::to_object::h68dec2fa6b9433d4Psi
  │    7:        0x101e04043 - dom::bindings::conversions::root_from_handlevalue::h520814305409971978
  │    8:        0x101faeebf - dom::bindings::conversions::Root<T>.FromJSValConvertible::from_jsval::h15970435205764620847
  │    9:        0x101fae698 - conversions::Vec<T>.FromJSValConvertible::from_jsval::h2572058210925103758
  │   10:        0x101fadf91 - dom::bindings::codegen::Bindings::TestBindingBinding::passInterfaceSequence::__rust_abi
  │   11:        0x101fadee8 - dom::bindings::codegen::Bindings::TestBindingBinding::passInterfaceSequence::h163005d06ddf76f2vaV
  │   12:        0x10338c4f0 - CallJitMethodOp
  │   13:        0x101c7c6fc - dom::bindings::utils::generic_call::hace9ffbdc84fda35qdg
  │   14:        0x101c7d8fb - dom::bindings::utils::generic_method::__rust_abi
  │   15:        0x101c7d8c2 - dom::bindings::utils::generic_method::h2e75bf07d318f9cb2gg
  │   16:        0x10348a56b - _ZN2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstructE
  │   17:        0x10349b0c9 - _ZL9InterpretP9JSContextRN2js8RunStateE
  │   18:        0x103491aa9 - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
  │   19:        0x10348a7a7 - _ZN2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstructE
  │   20:        0x1037b4dba - _ZN2js8fun_callEP9JSContextjPN2JS5ValueE
  │   21:        0x10348a56b - _ZN2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstructE
  │   22:        0x10349b0c9 - _ZL9InterpretP9JSContextRN2js8RunStateE
  │   23:        0x103491aa9 - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
  │   24:        0x10348a7a7 - _ZN2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstructE
  │   25:        0x1037b53ec - _ZN2js9fun_applyEP9JSContextjPN2JS5ValueE
  │   26:        0x10348a56b - _ZN2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstructE
  │   27:        0x10349b0c9 - _ZL9InterpretP9JSContextRN2js8RunStateE
  │   28:        0x103491aa9 - _ZN2js9RunScriptEP9JSContextRNS_8RunStateE
  │   29:        0x1034a0926 - _ZN2js13ExecuteKernelEP9JSContextN2JS6HandleIP8JSScriptEER8JSObjectRKNS2_5ValueENS_11ExecuteTypeENS_16AbstractFramePtrEPS9_
  │   30:        0x1034a0a53 - _ZN2js7ExecuteEP9JSContextN2JS6HandleIP8JSScriptEER8JSObjectPNS2_5ValueE
  │   31:        0x103780992 - _ZL8EvaluateP9JSContextN2JS6HandleIP8JSObjectEERKNS1_22ReadOnlyCompileOptionsERNS1_18SourceBufferHolderENS1_13MutableHandleINS1_5ValueEEE
  │   32:        0x103780afa - _ZN2JS8EvaluateEP9JSContextRKNS_22ReadOnlyCompileOptionsEPKDsmNS_13MutableHandleINS_5ValueEEE
  │   33:        0x101c621a1 - dom::window::_&'a T.ScriptHelpers::evaluate_script_on_global_with_result::h14423892878996675526
  │   34:        0x1021e9cf7 - dom::htmlscriptelement::HTMLScriptElement::execute::hcfa9911579ac35e0HbX
  │   35:        0x1023497d8 - dom::htmlscriptelement::HTMLScriptElement::prepare::h07ea3c0fa9162175a1W
  │   36:        0x1023d9ab3 - parse::html::servohtmlparser..Sink.TreeSink::complete_script::he8db6d7d61f7b124l1h
  │   37:        0x1023beaaf - tree_builder::rules::super..TreeBuilder<Handle, Sink>.TreeBuilderStep::step::h10164694725958941600
  │   38:        0x1023a43e3 - tree_builder::TreeBuilder<Handle, Sink>::process_to_completion::h17153553693321496153
  │   39:        0x10239f874 - tree_builder::TreeBuilder<Handle, Sink>.TokenSink::process_token::h3178721526101268622
  │   40:        0x10239ea3d - tokenizer::Tokenizer<Sink>::process_token::h2694761395103313964
  │   41:        0x1023ec593 - tokenizer::Tokenizer<Sink>::emit_current_tag::h12242332108089814736
  │   42:        0x1023e42cb - tokenizer::Tokenizer<Sink>::step::h3138513044164269277
  │   43:        0x1023e0691 - tokenizer::Tokenizer<Sink>::run::h12471341708133716696
  │   44:        0x10239562f - dom::servohtmlparser::ServoHTMLParser::parse_sync::hf5578181311529356o6
  │   45:        0x102394f7a - dom::servohtmlparser::ServoHTMLParser::resume::h99088d77b2e9cebbYq6
  │   46:        0x1021e5224 - dom::servohtmlparser::ParserRef<'a>::resume::h8519c327cbfb13acH45
  │   47:        0x101c246b8 - dom::document::Document::finish_load::hb79b6bd81805b978orG
  │   48:        0x1023462ae - dom::htmlscriptelement::ScriptContext.AsyncResponseListener::response_complete::hb57b555d98abc87eFZW
  │   49:        0x10285ad4b - ResponseAction::process::h56f886bfeb835e0cy1e
  │   50:        0x10234d4c5 - network_listener::ListenerRunnable<T>.Runnable::handler::h4398273096495339226
  │   51:        0x102521831 - script_thread::ScriptThread::handle_msg_from_script::h7880f3ae47c6e7eeNIj
  │   52:        0x1025a356f - script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::closure.174804
  │   53:        0x1025a313d - script_thread::ScriptThread::profile_event::h15894588503435751119
  │   54:        0x102575593 - script_thread::ScriptThread::handle_msgs::hfbbe208a039f529eruj
  │   55:        0x10251dbac - script_thread::ScriptThread::start::hb3498496f32d2febhuj
  │   56:        0x10251db64 - script_thread::ScriptThread.ScriptThreadFactory::create::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::closure.173544
  │   57:        0x10251d8d0 - mem::ProfilerChan::run_with_memory_reporting::h11282657348641104726
  │   58:        0x102506d26 - script_thread::ScriptThread.ScriptThreadFactory::create::_$u7b$$u7b$closure$u7d$$u7d$::closure.172924
  │   59:        0x102506117 - util::thread::spawn_named_with_send_on_failure::_$u7b$$u7b$closure$u7d$$u7d$::closure.172911
  │   60:        0x102505b03 - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::closure.172900
  │   61:        0x102505a78 - sys_common::unwind::try::try_fn::h6293295293492247684
  │   62:        0x103a5ff9b - __rust_try
  │   63:        0x103a5d023 - sys_common::unwind::inner_try::h3c719a7bb47af1af18s
  │   64:        0x1025059b5 - sys_common::unwind::try::h18179677261542290039
  │   65:        0x1025057ee - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::closure.172897
  │   66:        0x102506337 - boxed::F.FnBox<A>::call_box::h4217930612223354289
  │   67:        0x103a61ddd - sys::thread::Thread::new::thread_start::hbf136943ebd5e19atrx
  │   68:     0x7fff8845ac12 - _pthread_body
  │   69:     0x7fff8845ab8f - _pthread_start
  │ thread 'ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'assertion failed: *self.stack == mem::transmute(&*self)', /Users/nox/src/servo/.cargo/git/checkouts/rust-mozjs-ebb4917e843c0a11/master/src/rust.rs:334
  │ stack backtrace:
  │    1:        0x103a609f8 - sys::backtrace::tracing::imp::write::h268482af15af0a80Y8t
  │    2:        0x103a62b55 - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.42271
  │    3:        0x103a62756 - panicking::default_handler::h97bc664f7e2bfeb5Ocy
  │    4:        0x103a4e716 - sys_common::unwind::begin_unwind_inner::h98c0975c9b6b2fa0Zbt
  │    5:        0x101ba5427 - sys_common::unwind::begin_unwind::h3966960370320578234
  │    6:        0x101bfc3a0 - rust::Rooted<T>.Drop::drop::h5353961944728302588
  │    7:        0x101bfc2fc - js..jsapi_macos_64..Rooted<js..jsapi_macos_64..Value>::drop.133591::haddb7649724b9029
  │    8:        0x1021e9b99 - dom::htmlscriptelement::HTMLScriptElement::execute::hcfa9911579ac35e0HbX
  │    9:        0x1023497d8 - dom::htmlscriptelement::HTMLScriptElement::prepare::h07ea3c0fa9162175a1W
  │   10:        0x1023d9ab3 - parse::html::servohtmlparser..Sink.TreeSink::complete_script::he8db6d7d61f7b124l1h
  │   11:        0x1023beaaf - tree_builder::rules::super..TreeBuilder<Handle, Sink>.TreeBuilderStep::step::h10164694725958941600
  │   12:        0x1023a43e3 - tree_builder::TreeBuilder<Handle, Sink>::process_to_completion::h17153553693321496153
  │   13:        0x10239f874 - tree_builder::TreeBuilder<Handle, Sink>.TokenSink::process_token::h3178721526101268622
  │   14:        0x10239ea3d - tokenizer::Tokenizer<Sink>::process_token::h2694761395103313964
  │   15:        0x1023ec593 - tokenizer::Tokenizer<Sink>::emit_current_tag::h12242332108089814736
  │   16:        0x1023e42cb - tokenizer::Tokenizer<Sink>::step::h3138513044164269277
  │   17:        0x1023e0691 - tokenizer::Tokenizer<Sink>::run::h12471341708133716696
  │   18:        0x10239562f - dom::servohtmlparser::ServoHTMLParser::parse_sync::hf5578181311529356o6
  │   19:        0x102394f7a - dom::servohtmlparser::ServoHTMLParser::resume::h99088d77b2e9cebbYq6
  │   20:        0x1021e5224 - dom::servohtmlparser::ParserRef<'a>::resume::h8519c327cbfb13acH45
  │   21:        0x101c246b8 - dom::document::Document::finish_load::hb79b6bd81805b978orG
  │   22:        0x1023462ae - dom::htmlscriptelement::ScriptContext.AsyncResponseListener::response_complete::hb57b555d98abc87eFZW
  │   23:        0x10285ad4b - ResponseAction::process::h56f886bfeb835e0cy1e
  │   24:        0x10234d4c5 - network_listener::ListenerRunnable<T>.Runnable::handler::h4398273096495339226
  │   25:        0x102521831 - script_thread::ScriptThread::handle_msg_from_script::h7880f3ae47c6e7eeNIj
  │   26:        0x1025a356f - script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::closure.174804
  │   27:        0x1025a313d - script_thread::ScriptThread::profile_event::h15894588503435751119
  │   28:        0x102575593 - script_thread::ScriptThread::handle_msgs::hfbbe208a039f529eruj
  │   29:        0x10251dbac - script_thread::ScriptThread::start::hb3498496f32d2febhuj
  │   30:        0x10251db64 - script_thread::ScriptThread.ScriptThreadFactory::create::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::closure.173544
  │   31:        0x10251d8d0 - mem::ProfilerChan::run_with_memory_reporting::h11282657348641104726
  │   32:        0x102506d26 - script_thread::ScriptThread.ScriptThreadFactory::create::_$u7b$$u7b$closure$u7d$$u7d$::closure.172924
  │   33:        0x102506117 - util::thread::spawn_named_with_send_on_failure::_$u7b$$u7b$closure$u7d$$u7d$::closure.172911
  │   34:        0x102505b03 - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::closure.172900
  │   35:        0x102505a78 - sys_common::unwind::try::try_fn::h6293295293492247684
  │   36:        0x103a5ff9b - __rust_try
  │   37:        0x103a5d023 - sys_common::unwind::inner_try::h3c719a7bb47af1af18s
  │   38:        0x1025059b5 - sys_common::unwind::try::h18179677261542290039
  │   39:        0x1025057ee - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::closure.172897
  │   40:        0x102506337 - boxed::F.FnBox<A>::call_box::h4217930612223354289
  │   41:        0x103a61ddd - sys::thread::Thread::new::thread_start::hbf136943ebd5e19atrx
  │   42:     0x7fff8845ac12 - _pthread_body
  │   43:     0x7fff8845ab8f - _pthread_start
  └ thread panicked while panicking. aborting.

Ran 1 tests finished in 3.0 seconds.
  • 0 ran as expected. 0 tests skipped.
  • 1 tests crashed unexpectedly
```

File: tests/wpt/mozilla/tests/mozilla/sequence-hole.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!doctype html>
+<meta charset="utf-8">
+<title></title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+test(function() {
+  var t = new TestBinding;
+  assert_throws(new TypeError, function () { t.passInterfaceSequence([new Blob,, new Blob]) });
+}, "Sequences with holes are correctly handled");
```
Comment (Ms2ger) [inline tests/wpt/mozilla/tests/mozilla/sequence-hole.html]: What?

Also check `passSequence` and `passStringSequence`.

**Comment by nox** [comment]
Problem solved, ready to review.

File: tests/wpt/mozilla/tests/mozilla/sequence-hole.html (hunk 1 above)
Comment (nox) [inline tests/wpt/mozilla/tests/mozilla/sequence-hole.html]: These don't fail though, right?

**Comment by Ms2ger** [comment]
At the very least, we should document the contract of the various functions as to how they handle non-objects.

**Comment by Ms2ger** [comment]
Sorry, I forgot this earlier. Please also add the following test; r=me with that.
```js

test(function() {
  Object.defineProperty(Array.prototype, 1, { get: function() { throw 7; } });
  var test_error = { name: "test" };
  var t = new TestBinding;
  assert_throws(test_error, function() {
    t.passSequence([1,, 3]);
  };
  assert_throws(test_error, function() {
    t.passStringSequence(["camembert",, "comté"]);
  };
  assert_throws(test_error, function() {
    t.passInterfaceSequence([new Blob,, new Blob]);
  };
}, "Sequences with holes and a getter on the prototype are correctly handled");
```

**Comment by jdm** [comment]
@bors-servo: retry
* #9063

**Pull request merged**